src/Features/Core/Portable/Completion/CompletionService.cs

```diff
@@ -135,7 +135,7 @@ public virtual TextSpan GetDefaultCompletionListSpan(SourceText text, int caretP
             ImmutableArray<CompletionItem> items,
             string filterText)
         {
-            var helper = CompletionHelper.GetHelper(document);
```

Comment:
couldn't you just update this method to do exactly what you've expanded out all the code to do?